packages/cloud-functions/src/circuitBreaker/circuitBreaker.ts

```diff
@@ -0,0 +1,88 @@
+import * as functions from 'firebase-functions'
+import { KeyManagementServiceClient, protos } from '@google-cloud/kms'
+import express from 'express'
+import { CIRCUIT_BREAKER_KEY_PATH } from '../config'
+
+type CryptoKeyVersionState = keyof typeof protos.google.cloud.kms.v1.CryptoKeyVersion.CryptoKeyVersionState
+
+const client = new KeyManagementServiceClient()
+
+const app = express()
+app.use(express.json())
+
+async function getKeyState(): Promise<CryptoKeyVersionState> {
+  const getCryptoKeyVersionResult = await client.getCryptoKeyVersion({
+    name: CIRCUIT_BREAKER_KEY_PATH,
+  })
+  return getCryptoKeyVersionResult[0].state as CryptoKeyVersionState
+}
+
+app.get('/health', async (req, res) => {
+  return res.status(200).json({
+    status: 'ok',
+  })
+})
+
+app.get('/status', async (req, res) => {
+  const keyState = await getKeyState()
+  switch (keyState) {
+    case 'ENABLED':
+      return res.status(200).json({
+        status: 'enabled',
+      })
+    case 'DISABLED':
+      return res.status(200).json({
+        status: 'temporarily disabled',
+      })
+    case 'DESTROYED':
+      return res.status(200).json({
+        status: 'permanently disabled',
+      })
+    default:
+      console.error(`Unexpected key state: ${keyState}`)
+      return res.status(200).json({
+        status: 'unknown',
+      })
+  }
+})
+
+app.post('/unwrap-key', async (req, res) => {
+  const ciphertext = req.body?.ciphertext
+  if (!ciphertext) {
+    return res.status(400).json({
+      error: '"ciphertext" parameter must be provided',
+    })
+  }
```

Comment:
Input `ciphertext` should be an encoded buffer, and it seems implied here that it is `base64`. I'd recommend we make this explicit and check upfront that this is the case, especially because `Buffer.from(buffer, 'base64')` will output an empty buffer if the input is not valid base64 (I know, wtf).
```suggestion
  const BASE64_REGEXP = /^(?:[A-Za-z0-9+\/]{4})*(?:[A-Za-z0-9+\/]{2}==|[A-Za-z0-9+\/]{3}=)?$/
  const ciphertext = req.body?.ciphertext
  if (!ciphertext) {
    return res.status(400).json({
      error: '"ciphertext" parameter must be provided',
    })
  } else if (!BASE64_REGEXP.test(ciphertext)) {
    return res.status(400).json({
      error: '"ciphertext" parameter must be a base64 encoded buffer',
    })
  }
```

Reply:
Yeah, it should be `base64` encoded. Good catch on adding an explicit test for that. Will add that in! (also definitely wtf re: `Buffer.from` behavior lol)